src/app/public/modules/auth-http-client/local-endpoint-checker.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  HttpClient,
+  HttpHandler
+} from '@angular/common/http';
+
+import {
+  BBAuthClientFactory
+} from '@skyux/auth-client-factory';
+
+export class LocalEndpointChecker {
+
+  public static getUrlForLocalServe(requestUrl: string,
+                                    handler: HttpHandler,
+                                    baseLocalServer: string,
+                                    fallBack: () => Promise<string>): Promise<string> {
+    let urlData = BBAuthClientFactory.BBAuth.extractUrl(requestUrl);
+    let localPort = urlData.port;
+    if (localPort) {
+      let client: HttpClient = new HttpClient(handler);
+      let url = `${baseLocalServer}:${localPort}/monitor`;
+      return client.get(url).toPromise()
+        .then(() => {
+          return Promise.resolve(`${baseLocalServer}:${localPort}/${urlData.endpoint}`);
+        })
+        .catch(() => {
+          // TODO something here in order to debug not hitting local when you think you should be?
```

Comment:
perhaps just a `console.warn`? i _think_ that's a pattern that happens already for these kind of scenarios.

Reply:
Where are other examples of this?  I figured we had something but I didn't want to make assumptions.

Reply:
hmmm. Looks like [`SkyLogService`](https://github.com/search?q=org%3Ablackbaud+SkyLogService&type=Code) is only used in one place: [`SkyRepeaterItem`](https://github.com/blackbaud/skyux-lists/blob/9f46c12fec72fbe754e2eee91ffd71fe27bbc9ae/src/app/public/modules/repeater/repeater-item.component.ts#L365). so maybe we don't have an _established_ pattern for this. 🤷‍♂

Reply:
🤷‍♂